pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -15,438 +15,6 @@ Template for each `dtype` helper function using 1-d template
-
-    K, N = (<object> values).shape
-
-    # GH 2778
-    if N == 0:
-        return
-
-    if limit is None:
-        lim = N
-    else:
-        if not util.is_integer_object(limit):
-            raise ValueError('Limit must be an integer')
-        if limit < 1:
-            raise ValueError('Limit must be greater than 0')
-        lim = limit
-
-    for j in range(K):
-        fill_count = 0
-        val = values[j, N - 1]
-        for i in range(N - 1, -1, -1):
-            if mask[j, i]:
-                if fill_count >= lim:
-                    continue
-                fill_count += 1
-                values[j, i] = val
-            else:
-                fill_count = 0
-                val = values[j, i]
-
-
-@cython.boundscheck(False)
-@cython.wraparound(False)
-def is_monotonic_{{name}}(ndarray[{{c_type}}] arr, bint timelike):
-    """
-    Returns
-    -------
-    is_monotonic_inc, is_monotonic_dec, is_unique
-    """
-    cdef:
-        Py_ssize_t i, n
-        {{c_type}} prev, cur
-        bint is_monotonic_inc = 1
-        bint is_monotonic_dec = 1
-        bint is_unique = 1
-
-    n = len(arr)
-
-    if n == 1:
-        if arr[0] != arr[0] or (timelike and <int64_t>arr[0] == iNaT):
-            # single value is NaN
-            return False, False, True
-        else:
-            return True, True, True
-    elif n < 2:
-        return True, True, True
-
-    if timelike and <int64_t>arr[0] == iNaT:
-        return False, False, True
-
-    {{nogil_str}}
-    {{tab}}prev = arr[0]
-    {{tab}}for i in range(1, n):
-    {{tab}}    cur = arr[i]
-    {{tab}}    if timelike and <int64_t>cur == iNaT:
-    {{tab}}        is_monotonic_inc = 0
-    {{tab}}        is_monotonic_dec = 0
-    {{tab}}        break
-    {{tab}}    if cur < prev:
-    {{tab}}        is_monotonic_inc = 0
-    {{tab}}    elif cur > prev:
-    {{tab}}        is_monotonic_dec = 0
-    {{tab}}    elif cur == prev:
-    {{tab}}        is_unique = 0
-    {{tab}}    else:
-    {{tab}}        # cur or prev is NaN
-    {{tab}}        is_monotonic_inc = 0
-    {{tab}}        is_monotonic_dec = 0
-    {{tab}}        break
-    {{tab}}    if not is_monotonic_inc and not is_monotonic_dec:
-    {{tab}}        is_monotonic_inc = 0
-    {{tab}}        is_monotonic_dec = 0
-    {{tab}}        break
-    {{tab}}    prev = cur
-    return is_monotonic_inc, is_monotonic_dec, \
-           is_unique and (is_monotonic_inc or is_monotonic_dec)
-
-
-@cython.wraparound(False)
-@cython.boundscheck(False)
-def arrmap_{{name}}(ndarray[{{c_type}}] index, object func):
-    cdef:
-        Py_ssize_t length = index.shape[0]
-        Py_ssize_t i = 0
-        ndarray[object] result = np.empty(length, dtype=np.object_)
-
-    from pandas._libs.lib import maybe_convert_objects
-
-    for i in range(length):
-        result[i] = func(index[i])
-
-    return maybe_convert_objects(result)
-
-{{endfor}}
-
 #----------------------------------------------------------------------
```

Comment:
what's left here? possible to fully remove this file?

Reply:
AFAICT diff_2d_dtpye is not straightforward to using fused types